# **Import Libraries**

In [2]:
import numpy as np
import pandas as pd

## **Loading File**

In [3]:
df = pd.read_excel("E:/project1_of_decodelab/Dataset for Data Analytics.xlsx")

# **Initial Inspection**

In [4]:
df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [5]:
df.shape

(1200, 14)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

There are some missing values in CouponCode column and we will fix it in next project

In [8]:
df.describe()

,Date,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200,1200.000000,1200.000000,1200.000000,1200.000000
mean,2024-03-22 16:58:48,2.945833,356.412750,5.485000,1053.968300
min,2023-01-01 00:00:00,1.000000,11.390000,1.000000,11.390000
25%,2023-08-03 18:00:00,2.000000,186.062500,4.000000,410.520000
50%,2024-03-23 00:00:00,3.000000,364.210000,5.000000,823.615000
75%,2024-11-08 12:00:00,4.000000,521.570000,7.000000,1578.475000
max,2025-06-30 00:00:00,5.000000,699.930000,10.000000,3456.400000
std,NaN,1.407557,197.177146,2.281983,819.856558


## **Checking Duplicates**

In [9]:
duplicate_count = df.duplicated().sum()
print(f"Total Duplicate Rows: {duplicate_count}")

Total Duplicate Rows: 0


In [10]:
id_duplicates = df.duplicated(subset=['OrderID']).sum()
print(f"Duplicate Order IDs: {id_duplicates}")

Duplicate Order IDs: 0


## **Checking Data Types**

In [11]:
df.dtypes

OrderID                    object
Date               datetime64[ns]
CustomerID                 object
Product                    object
Quantity                    int64
UnitPrice                 float64
ShippingAddress            object
PaymentMethod              object
OrderStatus                object
TrackingNumber             object
ItemsInCart                 int64
CouponCode                 object
ReferralSource             object
TotalPrice                float64
dtype: object

# **Missing Value Analysis**

In [12]:
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

# Check Text Column

In [14]:
text_cols = [
    'Product',
    'PaymentMethod',
    'OrderStatus',
    'CouponCode',
    'ReferralSource'
]

for col in text_cols:
    print(df[col].unique())

['Monitor' 'Phone' 'Tablet' 'Chair' 'Printer' 'Laptop' 'Desk']
['Debit Card' 'Online' 'Credit Card' 'Gift Card' 'Cash']
['Shipped' 'Cancelled' 'Returned' 'Delivered' 'Pending']
['SAVE10' 'FREESHIP' nan 'WINTER15']
['Instagram' 'Referral' 'Email' 'Facebook' 'Google']


## Clean Text

In [20]:
for col in text_cols:
    df[col] = df[col].str.strip().str.title()

## **Data Validation Checks**

In [23]:
print((df['UnitPrice'] <= 0).sum())
print((df['TotalPrice'] <= 0).sum())

0
0


In [29]:
(
    df['CalculatedTotal'].round(2) !=
    df['TotalPrice'].round(2)
).sum()
df[['Quantity', 'UnitPrice', 'TotalPrice', 'CalculatedTotal']].head()


,Quantity,UnitPrice,TotalPrice,CalculatedTotal
0,5,570.62,2853.10,2853.10
1,2,151.35,302.70,302.70
2,5,550.68,2753.40,2753.40
3,1,273.19,273.19,273.19
4,4,626.01,2504.04,2504.04


In [30]:
mismatch = df[
    df['CalculatedTotal'].round(2) !=
    df['TotalPrice'].round(2)
]

print("Mismatched Rows:", len(mismatch))

Mismatched Rows: 0


### Tasks Performed

* Checked dataset structure and data types
* Identified missing values
* Verified duplicate records
* Examined data consistency
* Prepared dataset for analysis

### Key Findings

* Dataset contained 1,200 rows and 14 columns.
* No duplicate rows were found.
* No duplicate Order IDs were detected.
* Coupon_Code contained 309 missing values.
* All remaining columns were complete.
* Missing coupon values indicate many customers placed orders without promotional offers.
* Dataset quality was high and required minimal preprocessing.